### DISCUSSION TITLE
Intermittent unresolvable dependency when used with Laravel Octane

### DISCUSSION THREAD

**Discussion by gxgstan** [livewire/livewire#3014] [7. Bugs (archived)] [answered]
### Description

I get this error from time to time only with Livewire components. Not sure how to reproduce it or what causes it.

```
Illuminate\Contracts\Container\BindingResolutionException
Unresolvable dependency resolving [Parameter #0 [ <required> $methods ]] in class Illuminate\Routing\Route
```

**Context**
- Livewire version: 2.4.4
- Laravel version: 8
- Octane version: 1.0.0
- Browser: All

**Comment by richan-fongdasen** [comment]
I also get the same intermittent error. At first, switching between Livewire routes didn't produce any error. The error occurs mostly when I move to a different Livewire route after updating the Livewire component's state a few times on the previous page.

**Context**

- Livewire version: 2.4.4
- Laravel framework version: 8.41.0
- Octane version: 1.0.0 (swoole)

**Comment by ariaieboy** [comment]
same issue here i use full-page component for my routes and randomly when i navigate between routes its show me this error and i need to force reload using (Ctrl + F5) the page multiple time so i can acces to the routes. or reload the octane workers
the weird thing is that when this errors show's up i can't acces to any other route that use full-page component i think its a memory leak or sth in the core of livewire

@calebporzio can you check this out with octane i can giveyou the code that reproduce the same problem.

**Comment by masterix21** [comment]
Same issue here, using routes with livewire components directly (`Route::get('/test', LivewireComponent::class)`).

In my case seems to be related to the `/livewire/message/media-library` POST call, makes from Spatie Laravel Medialibrary Pro.

Are you using Spatie's package?

**Context**

- Livewire version: 2.5.5
- Laravel version: 8.52.0
- Octane version: 1.0.8 (swoole)

**Reply by ariaieboy** [reply]
i have this problem and i am not using any spatie package.
i tested it on a fresh install of laravel-livewire without any scaffoliding and still i get this message.
the only work around is to use artisan optimize.

**Reply by masterix21** [reply]
Yes, you're right: I confirm that **doesn't** depend due to a bug in the Spatie package directly, because removing the Spatie package the issue disappeared, but sometimes still appears with no logic, and disappears restarting Octane.

**Comment by joshhanley** [comment, marked as answer]
This has been fixed in https://github.com/livewire/livewire/pull/3946 and should be in the next release of Livewire.

Hope this helps!

**Reply by ariaieboy** [reply]
yes in my project after pulling this PR i didnt faced this error anymore i hope this 1 line change work for real and end this annoying error
